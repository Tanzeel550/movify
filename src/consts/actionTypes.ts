export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthActionsType = {
  type: 'LOGIN' | 'LOGOUT';
  user?: object;
};

export type AuthUserType = {
  user: object;
};

export type MoviesActionReturnType = {
  type: 'CREATE_MOVIE' | 'GET_MOVIES' | 'UPDATE_MOVIE' | 'DELETE_MOVIE';
  movie?: object;
  movies?: object;
  id?: string;
};

export type MoviesActionParamsType = {
  id?: string;
  movie?: object;
  movies?: object;
};

export type SEARCH_TYPE = {
  text: string;
};

export type ErrorType = {
  message: string;
  title: string;
};

export type ErrorActionsReturnType = {
  type: 'SET_ERROR' | 'CLEAR_ERROR';
  message?: string;
  title?: string;
};
