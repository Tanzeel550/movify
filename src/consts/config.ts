const OMBD_API_KEY = process.env.REACT_APP_OMDB_API_KEY;
export const SEARCH_MOVIE_BY_TEXT = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&s=`;
export const GET_MOVIE_BY_TITLE = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&t=`;
export const GET_MOVIE_BY_IMDB_ID = `https://www.omdbapi.com/?apikey=${OMBD_API_KEY}&i=`;

export const PROMISE_AREAS = {
  SEARCH_MOVIE_BY_TEXT: 'SEARCH_MOVIE_BY_TEXT',
  GET_MOVIE_BY_TITLE: 'GET_MOVIE_BY_TITLE',
  GET_ALL_MOVIES: 'GET_ALL_MOVIES',
  LOAD_PAGE: 'LOAD_PAGE',
};

export const skills1 = [
  'Web Design',
  'JavaScript Developer',
  'React JS',
  'Rest API',
  'NodeJS',
];

export const skills2 = ['Company', 'Team', 'Careers'];

export const TEST_MOVIES_SRC = 'movies_test';
