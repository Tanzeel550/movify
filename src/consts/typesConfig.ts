export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthActionsType = {
  type: 'LOGIN' | 'LOGOUT';
  user?: object;
};

export type AuthUserType = {
  user: object;
};

export type MOVIES_ACTION = {
  type: 'CREATE_MOVIE' | 'GET_MOVIES' | 'UPDATE_MOVIE' | 'DELETE_MOVIE';
  movie?: object;
  movies?: object;
  id?: string;
};

export type SEARCH_TYPE = {
  text: string;
};

export type MOVIE_ID = {
  id?: string;
  movie?: object;
  movies?: object;
};
