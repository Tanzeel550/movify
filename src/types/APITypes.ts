// ============================================================
// ======================== API Types =========================
// ============================================================

export type SEARCH_TYPE = {
  text: string;
};

export type APISingleMovieResultBySearch = {
  Title: string;
  Year: string;
  imdbID: string;
  Type: string;
  Poster: string;
};

export type APIMovieResultsBySearch = APISingleMovieResultBySearch[];

export type Rating = {
  Source: string;
  Value: string;
};

export type APIMovieResultByTitle = {
  Title?: string;
  Year?: string;
  Rated?: string;
  Released?: string;
  Runtime?: string;
  Genre?: string;
  Director?: string;
  Writer?: string;
  Actors?: string;
  Plot?: string;
  Language?: string;
  Country?: string;
  Awards?: string;
  Poster?: string;
  Ratings?: Rating[];
  Metascore?: string;
  imdbRating?: string;
  imdbVotes?: string;
  imdbID?: string;
  Type?: string;
  DVD?: string;
  BoxOffice?: string;
  Production?: string;
  Website?: string;
  Response?: string;
};

export type FireDBMovieItem = {
  id?: string;
  name: string;
  watched: boolean;
  dateWatched: any;
  whatYouLearnt: string;
  Poster: string;
  createdAt: number;
};
