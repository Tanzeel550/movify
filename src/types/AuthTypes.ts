// ============================================================
// ======================== Auth Types ========================
// ============================================================

export type AuthUserType = {
  user: object;
};

export type EmailPassType = {
  email: string;
  password: string;
};

export type AuthDefaultType = {
  isAuthenticated: boolean;
  user: object;
};

export type TestUserType = {
  uid: string;
  name: string;
  email: string;
  passion: string;
  os: string;
  version: string;
  height: number;
};
