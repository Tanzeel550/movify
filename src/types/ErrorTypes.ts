// ============================================================
// ======================= Error Types ========================
// ============================================================

export type ErrorState = {
  message: string;
  title: string;
};

export type ErrorActionsReturnType = {
  type: 'SET_ERROR' | 'CLEAR_ERROR';
  message?: string;
  title?: string;
};
