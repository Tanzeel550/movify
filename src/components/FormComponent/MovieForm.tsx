import React, { useEffect, useState } from 'react';

import {
  getMovieByTitle,
  searchMovieByText,
} from '../../actions/moviesAsyncActions';
import SearchComponent from './SearchComponent';
import PromiseLoading from '../Utils/PromiseLoading';
import MovieDetails from './MovieDetails';
import { trackPromise } from 'react-promise-tracker';
import { PROMISE_AREAS } from '../../consts/config';
import { APIMovieResultByTitle, FireDBMovieItem } from '../../types/APITypes';

type Props = {
  movie?: FireDBMovieItem;
  actionType: 'update' | 'add';
  handleFormSubmit: (data: FireDBMovieItem) => {};
  handleMovieDeletion?: () => {};
};

const MovieForm: React.FC<Props> = ({
  movie,
  actionType,
  handleFormSubmit,
  handleMovieDeletion,
}: Props) => {
  const formatDate = (date: any): string => {
    const dateObj = new Date(date);
    const day = `${dateObj.getDate()}`.padStart(2, '0');
    const mm = `${dateObj.getMonth() + 1}`.padStart(2, '0');
    const yyyy = dateObj.getFullYear();
    return `${yyyy}-${mm}-${day}`;
  };

  const [todayDate] = useState(formatDate(Date.now()));
  const [name, setName] = useState(movie?.name || '');
  const [watched, setWatched] = useState(movie?.watched || true);
  const [dateWatched, setDateWatched] = useState(
    movie?.dateWatched || todayDate
  );
  const [whatYouLearnt, setWhatYouLearnt] = useState(
    movie?.whatYouLearnt || ''
  );
  const [Poster, setPoster] = useState(movie?.Poster || '');
  const [createdAt] = useState(movie?.createdAt || Date.now() + '');
  const [updatedAt] = useState(Date.now());

  const [searchedMovies, setSearchedMovies] = useState([]);
  const [selectedMovie, setSelectedMovie] = useState<
    APIMovieResultByTitle | undefined
  >();

  useEffect(() => {
    (async () => {
      document
        .getElementById('dateWatched')!
        .setAttribute('max', formatDate(new Date()));
      if (movie) {
        const selectedMovie: APIMovieResultByTitle = (await trackPromise(
          getMovieByTitle({ text: name }),
          PROMISE_AREAS.GET_MOVIE_BY_TITLE
        )) as APIMovieResultByTitle;
        setSelectedMovie(selectedMovie);
      }
    })();
  }, [movie, name]);

  const handleSearchItemClick = async (title: string): Promise<void> => {
    setSelectedMovie(undefined);
    const selectedMovie: APIMovieResultByTitle = (await trackPromise(
      getMovieByTitle({ text: title }),
      PROMISE_AREAS.GET_MOVIE_BY_TITLE
    )) as APIMovieResultByTitle;
    setSearchedMovies([]);
    setName(selectedMovie.Title!);
    setPoster(selectedMovie.Poster!);
    setSelectedMovie(selectedMovie);
  };

  const localFormSubmit = (e: React.FormEvent): void => {
    e.preventDefault();
    const movie = {
      name: name,
      watched: watched,
      dateWatched: dateWatched,
      whatYouLearnt: whatYouLearnt,
      Poster: Poster,
      createdAt: createdAt + '',
      updatedAt: updatedAt + '',
    };
    handleFormSubmit(movie);
  };

  const handleNameChange = async (
    e: React.ChangeEvent<HTMLInputElement>
  ): Promise<void> => {
    await setName(e.target.value);
    await setSearchedMovies([]);
    // @ts-ignore
    const searchedMovies: APIMovieResultsBySearch = (await trackPromise(
      searchMovieByText({ text: name }),
      PROMISE_AREAS.SEARCH_MOVIE_BY_TEXT
    )) as APIMovieResultByTitle;
    if (!searchedMovies) return;
    setSearchedMovies(searchedMovies.slice(0, 5));
  };

  const handleDateChange = (e: React.ChangeEvent<HTMLInputElement>): void => {
    setDateWatched(formatDate(e.target.value));
  };

  const handleWatchedChange = (
    e: React.ChangeEvent<HTMLInputElement>
  ): void => {
    // @ts-ignore
    setWatched(Boolean(e.target.value));
  };

  const handleTextAreaChange = (
    e: React.ChangeEvent<HTMLTextAreaElement>
  ): void => {
    setWhatYouLearnt(e.target.value);
  };

  return (
    <div className="container">
      <div className="row movie__form">
        <div className="col-4">
          <img src={Poster} alt="" />
        </div>
        <div className="col-8">
          <form className="row" onSubmit={localFormSubmit}>
            <div className="col-sm-12 row">
              <div className="col-sm-10 movie__form--label-name">
                <label htmlFor="name">Name</label>
                <input
                  type="text"
                  id="name"
                  className="form-control"
                  value={name}
                  onChange={handleNameChange}
                />
              </div>
              <div className="col-sm-2">
                <label htmlFor="isWatched">Watched</label>
                <input
                  type="checkbox"
                  id="isWatched"
                  className="form-check"
                  value={`${watched}`}
                  defaultChecked={true}
                  onChange={handleWatchedChange}
                />
              </div>
            </div>
            {watched && (
              <div className="col-sm-12 row">
                <div className="col-lg-4 col-md-4">
                  <label htmlFor="dateWatched">Date Watched:</label>
                  <input
                    type="date"
                    id="dateWatched"
                    value={dateWatched}
                    onChange={handleDateChange}
                    className="form-control"
                  />
                </div>
                <div className="col-lg-8 col-md-8" />
                <div className="col-sm-12">
                  <label htmlFor="text-area">What You Learnt from it?</label>
                  <textarea
                    value={whatYouLearnt}
                    onChange={handleTextAreaChange}
                    id="text-area"
                    className="form-control"
                    style={{ height: 285 }}
                  />
                </div>
              </div>
            )}
            {actionType === 'add' && (
              <div
                className="col-sm-12 d-flex justify-content-center"
                onClick={localFormSubmit}
              >
                <button className="btn btn-primary btn-lg mt-3">Create</button>
              </div>
            )}
            {actionType === 'update' && (
              <div className="col-sm-12 d-flex justify-content-center">
                <button
                  className="btn btn-primary btn-lg mt-3"
                  onClick={localFormSubmit}
                >
                  Update
                </button>
                <button
                  className="btn btn-danger btn-lg mt-3"
                  onClick={handleMovieDeletion}
                >
                  Remove
                </button>
              </div>
            )}
            {searchedMovies?.length > 0 && (
              <div>
                <PromiseLoading area={PROMISE_AREAS.SEARCH_MOVIE_BY_TEXT} />
                <SearchComponent
                  handleSearchItemClick={handleSearchItemClick}
                  movies={searchedMovies}
                />
              </div>
            )}
          </form>
        </div>
        <div>
          <PromiseLoading area={PROMISE_AREAS.GET_MOVIE_BY_TITLE} />
          {selectedMovie && <MovieDetails movie={selectedMovie} />}
        </div>
      </div>
    </div>
  );
};

export default MovieForm;
