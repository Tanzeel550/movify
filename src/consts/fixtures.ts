import {
  APIMovieResultByTitle,
  APISingleMovieResultBySearch,
  FireDBMovieItem,
} from '../types/APITypes';
import { AuthUserType } from '../types/AuthTypes';

export const movies: FireDBMovieItem[] = [
  {
    id: '1234567',
    name: 'The Dark Knight',
    watched: true,
    dateWatched: '234567890',
    whatYouLearnt: 'The world is as good as we let it to be',
    Poster: 'https://image.com',
    createdAt: '9876543',
  },
  {
    id: '4567890',
    name: 'GodFather',
    watched: false,
    dateWatched: '98765434567',
    whatYouLearnt: 'You can act like a man what is the matter with you?',
    Poster: 'https://godfather.com',
    createdAt: '987654567890',
  },
  {
    id: '9ikjnbvfrty',
    name: 'Avengers: EndGame',
    watched: false,
    dateWatched: '',
    whatYouLearnt: 'Thanos will be defeated',
    Poster: 'https://avengers.com',
    createdAt: '69864123340404',
  },
];

export const myUser: AuthUserType = {
  user: {
    name: 'Tanzeel',
    email: 'tanzeelahmed550@gmail.com',
    passion: 'Web Dev',
    os: 'Kali Linux',
    version: '1.0.0',
    height: 5.5,
  },
};

export const skills1 = [
  'Web Design',
  'JavaScript Developer',
  'React JS',
  'Rest API',
  'NodeJS',
];

export const skills2 = ['Company', 'Team', 'Careers'];

export const email = 'unknownPerson123@uet.com';
export const password = 'JonSnow';
export const nameOfTheMovie = 'Godfather';

export const movieFromAPIByTitle: APIMovieResultByTitle = {
  Title: 'The Dark Knight',
  Year: '2008',
  Rated: 'PG-13',
  Released: '18 Jul 2008',
  Runtime: '152 min',
  Genre: 'Action, Crime, Drama, Thriller',
  Director: 'Christopher Nolan',
  Writer:
    'Jonathan Nolan (screenplay), Christopher Nolan (screenplay), Christopher Nolan (story), David S. Goyer (story), Bob Kane (characters)',
  Actors: 'Christian Bale, Heath Ledger, Aaron Eckhart, Michael Caine',
  Plot:
    'When the menace known as the Joker wreaks havoc and chaos on the people of Gotham, Batman must accept one of the greatest psychological and physical tests of his ability to fight injustice.',
  Language: 'English, Mandarin',
  Country: 'USA, UK',
  Awards: 'Won 2 Oscars. Another 157 wins & 163 nominations.',
  Poster:
    'https://m.media-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SX300.jpg',
  Ratings: [
    { Source: 'Internet Movie Database', Value: '9.0/10' },
    {
      Source: 'Rotten Tomatoes',
      Value: '94%',
    },
    { Source: 'Metacritic', Value: '84/100' },
  ],
  Metascore: '84',
  imdbRating: '9.0',
  imdbVotes: '2,341,244',
  imdbID: 'tt0468569',
  Type: 'movie',
  DVD: '14 Jun 2010',
  BoxOffice: '$534,858,444',
  Production: 'Syncopy',
  Website: 'N/A',
  Response: 'True',
};

export const moviesFromAPIBySearch: APISingleMovieResultBySearch[] = [
  {
    Title: 'The Dark Knight',
    Year: '2008',
    imdbID: 'tt0468569',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMTMxNTMwODM0NF5BMl5BanBnXkFtZTcwODAyMTk2Mw@@._V1_SX300.jpg',
  },
  {
    Title: 'The Dark Knight Rises',
    Year: '2012',
    imdbID: 'tt1345836',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMTk4ODQzNDY3Ml5BMl5BanBnXkFtZTcwODA0NTM4Nw@@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns, Part 1',
    Year: '2012',
    imdbID: 'tt2313197',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMzIxMDkxNDM2M15BMl5BanBnXkFtZTcwMDA5ODY1OQ@@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns, Part 2',
    Year: '2013',
    imdbID: 'tt2166834',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BYTEzMmE0ZDYtYWNmYi00ZWM4LWJjOTUtYTE0ZmQyYWM3ZjA0XkEyXkFqcGdeQXVyNTA4NzY1MzY@._V1_SX300.jpg',
  },
  {
    Title: 'Batman: The Dark Knight Returns',
    Year: '2013',
    imdbID: 'tt11060882',
    Type: 'movie',
    Poster:
      'https://m.media-amazon.com/images/M/MV5BMjExMGE4NDQtM2MwMS00YzhlLTlkYjgtNmExNzM1ZDBhYzg1XkEyXkFqcGdeQXVyMTY5Nzc4MDY@._V1_SX300.jpg',
  },
];
