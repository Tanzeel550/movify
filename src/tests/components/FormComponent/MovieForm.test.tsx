import { shallow, ShallowWrapper } from 'enzyme';
import MovieForm from '../../../components/FormComponent/MovieForm';
import React from 'react';
import { movies, nameOfTheMovie } from '../../../consts/fixtures';
import { FireDBMovieItem } from '../../../types/APITypes';

let wrapper: ShallowWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>,
  actionType: 'update' | 'add',
  handleFormSubmit: jest.Mock<any, any>,
  handleMovieDeletion: jest.Mock<any, any>;

beforeEach(() => {
  actionType = 'add';
  handleFormSubmit = jest.fn();
  wrapper = shallow(
    <MovieForm actionType={actionType} handleFormSubmit={handleFormSubmit} />
  );
});

test('should render MovieForm correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

const fuck = ({
  at,
  value,
  expected,
}: {
  at: number;
  value: any;
  expected: any;
}) => {
  wrapper.find('input').at(at).simulate('change', { target: { value } });
  expect(wrapper.find('input').at(at).prop('value')).toBe(expected);
};

test('should change name when nameOfTheMovie by the user', () => {
  fuck({ at: 0, value: nameOfTheMovie, expected: nameOfTheMovie });
});

test('should change when watched is changed by the user', () => {
  fuck({ at: 1, value: true, expected: 'true' });
});

test('should change when Date Watched is changed by the user', () => {
  fuck({ at: 2, value: '2021-06-01', expected: '2021-06-01' });
});

test('should change when What You Learnt is changed by the user', () => {
  wrapper.find('textarea').simulate('change', { target: { value: 'Gang' } });
  expect(wrapper.find('textarea').prop('value')).toBe('Gang');
});

test('should not render Date Watched and What You Learnt when watched is false', () => {
  fuck({ at: 1, value: false, expected: 'false' });
  expect(wrapper).toMatchSnapshot();
});

test('form should become update when actionType is update', () => {
  handleMovieDeletion = jest.fn();
  const movie = movies[0];
  wrapper = shallow(
    <MovieForm
      actionType="update"
      handleFormSubmit={handleFormSubmit}
      movie={movie}
      handleMovieDeletion={handleMovieDeletion}
    />
  );

  expect(wrapper).toMatchSnapshot();

  expect(wrapper.find('input').at(0).prop('value')).toBe(movie.name);
  expect(wrapper.find('input').at(1).prop('value')).toBe(`${movie.watched}`);
  expect(wrapper.find('input').at(2).prop('value')).toBe(movie.dateWatched);
  expect(wrapper.find('textarea').prop('value')).toBe(movie.whatYouLearnt);
  expect(wrapper.find('img').prop('src')).toBe(movie.Poster);

  wrapper.find('form').simulate('submit', {
    preventDefault: () => {},
  });
  const expectedMovie: FireDBMovieItem = {
    name: movie.name,
    watched: movie.watched,
    dateWatched: movie.dateWatched,
    whatYouLearnt: movie.whatYouLearnt,
    Poster: movie.Poster,
    createdAt: movie.createdAt,
    updatedAt: expect.any(String),
  };
  expect(handleFormSubmit).toBeCalledWith(expectedMovie);

  wrapper.find('button').at(1).simulate('click');
  expect(handleMovieDeletion).toBeCalledTimes(1);
});

export {};
