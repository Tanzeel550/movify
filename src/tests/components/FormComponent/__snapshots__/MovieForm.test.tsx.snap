// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`form should become update when actionType is update 1`] = `
<div
  className="container"
>
  <div
    className="row movie__form"
  >
    <div
      className="col-4"
    >
      <img
        alt=""
        src="https://image.com"
      />
    </div>
    <div
      className="col-8"
    >
      <form
        className="row"
        onSubmit={[Function]}
      >
        <div
          className="col-sm-12 row"
        >
          <div
            className="col-sm-10 movie__form--label-name"
          >
            <label
              htmlFor="name"
            >
              Name
            </label>
            <input
              className="form-control"
              id="name"
              onChange={[Function]}
              type="text"
              value="The Dark Knight"
            />
          </div>
          <div
            className="col-sm-2"
          >
            <label
              htmlFor="isWatched"
            >
              Watched
            </label>
            <input
              className="form-check"
              defaultChecked={true}
              id="isWatched"
              onChange={[Function]}
              type="checkbox"
              value="true"
            />
          </div>
        </div>
        <div
          className="col-sm-12 row"
        >
          <div
            className="col-lg-4 col-md-4"
          >
            <label
              htmlFor="dateWatched"
            >
              Date Watched:
            </label>
            <input
              className="form-control"
              id="dateWatched"
              onChange={[Function]}
              type="date"
              value="234567890"
            />
          </div>
          <div
            className="col-lg-8 col-md-8"
          />
          <div
            className="col-sm-12"
          >
            <label
              htmlFor="text-area"
            >
              What You Learnt from it?
            </label>
            <textarea
              className="form-control"
              id="text-area"
              onChange={[Function]}
              style={
                Object {
                  "height": 285,
                }
              }
              value="The world is as good as we let it to be"
            />
          </div>
        </div>
        <div
          className="col-sm-12 d-flex justify-content-center"
        >
          <button
            className="btn btn-primary btn-lg mt-3"
            onClick={[Function]}
          >
            Update
          </button>
          <button
            className="btn btn-danger btn-lg mt-3"
            onClick={[MockFunction]}
          >
            Remove
          </button>
        </div>
      </form>
    </div>
    <div>
      <PromiseLoading
        area="GET_MOVIE_BY_TITLE"
      />
    </div>
  </div>
</div>
`;

exports[`should not render Date Watched and What You Learnt when watched is false 1`] = `
<div
  className="container"
>
  <div
    className="row movie__form"
  >
    <div
      className="col-4"
    >
      <img
        alt=""
        src=""
      />
    </div>
    <div
      className="col-8"
    >
      <form
        className="row"
        onSubmit={[Function]}
      >
        <div
          className="col-sm-12 row"
        >
          <div
            className="col-sm-10 movie__form--label-name"
          >
            <label
              htmlFor="name"
            >
              Name
            </label>
            <input
              className="form-control"
              id="name"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <div
            className="col-sm-2"
          >
            <label
              htmlFor="isWatched"
            >
              Watched
            </label>
            <input
              className="form-check"
              defaultChecked={true}
              id="isWatched"
              onChange={[Function]}
              type="checkbox"
              value="false"
            />
          </div>
        </div>
        <div
          className="col-sm-12 d-flex justify-content-center"
          onClick={[Function]}
        >
          <button
            className="btn btn-primary btn-lg mt-3"
          >
            Create
          </button>
        </div>
      </form>
    </div>
    <div>
      <PromiseLoading
        area="GET_MOVIE_BY_TITLE"
      />
    </div>
  </div>
</div>
`;

exports[`should render MovieForm correctly 1`] = `
<div
  className="container"
>
  <div
    className="row movie__form"
  >
    <div
      className="col-4"
    >
      <img
        alt=""
        src=""
      />
    </div>
    <div
      className="col-8"
    >
      <form
        className="row"
        onSubmit={[Function]}
      >
        <div
          className="col-sm-12 row"
        >
          <div
            className="col-sm-10 movie__form--label-name"
          >
            <label
              htmlFor="name"
            >
              Name
            </label>
            <input
              className="form-control"
              id="name"
              onChange={[Function]}
              type="text"
              value=""
            />
          </div>
          <div
            className="col-sm-2"
          >
            <label
              htmlFor="isWatched"
            >
              Watched
            </label>
            <input
              className="form-check"
              defaultChecked={true}
              id="isWatched"
              onChange={[Function]}
              type="checkbox"
              value="true"
            />
          </div>
        </div>
        <div
          className="col-sm-12 row"
        >
          <div
            className="col-lg-4 col-md-4"
          >
            <label
              htmlFor="dateWatched"
            >
              Date Watched:
            </label>
            <input
              className="form-control"
              id="dateWatched"
              onChange={[Function]}
              type="date"
              value=""
            />
          </div>
          <div
            className="col-lg-8 col-md-8"
          />
          <div
            className="col-sm-12"
          >
            <label
              htmlFor="text-area"
            >
              What You Learnt from it?
            </label>
            <textarea
              className="form-control"
              id="text-area"
              onChange={[Function]}
              style={
                Object {
                  "height": 285,
                }
              }
              value=""
            />
          </div>
        </div>
        <div
          className="col-sm-12 d-flex justify-content-center"
          onClick={[Function]}
        >
          <button
            className="btn btn-primary btn-lg mt-3"
          >
            Create
          </button>
        </div>
      </form>
    </div>
    <div>
      <PromiseLoading
        area="GET_MOVIE_BY_TITLE"
      />
    </div>
  </div>
</div>
`;
